src/frequenz/sdk/model/_manager.py

```diff
@@ -0,0 +1,366 @@
+                between model keys and paths.
+        """
+        self._models: dict[Any, _Model[T]] = {}
+
+        models_config = model_repository.get_models_config()
+        for key, model_path in models_config.items():
+            self._models[key] = _Model(data=self._load(model_path), path=model_path)
+
+        self._monitoring_task: asyncio.Task[None] = asyncio.create_task(
+            self._start_model_monitor()
+        )
+
+    @classmethod
+    def single(cls, model_path: str) -> "ModelManager[T]":
+        """Create a model manager instance for a single model.
+
+        Args:
+            model_path: the path to the single model file.
+
+        Returns:
+            a model manager instance for a single model.
+        """
+        model_repository = SingleModelRepository(model_path=model_path)
+        return cls(model_repository=model_repository)
+
+    @classmethod
+    def weekday(cls, directory: str, file_name: str) -> "ModelManager[T]":
+        """Create a model manager instance for weekday-based models.
+
+        Args:
+            directory: the base directory where the weekday-based models are stored.
+            file_name: the name of the file containing the model.
+
+        Returns:
+            a model manager instance for weekday-based models.
+        """
+        model_repository = WeekdayModelRepository(
+            directory=directory, file_name=file_name
+        )
+        return cls(model_repository=model_repository)
+
+    @classmethod
+    def interval(cls, directory: str, file_name: str) -> "ModelManager[T]":
+        """Create a model manager instance for interval-based models.
+
+        Args:
+            directory: the base directory where the interval-based models are stored.
+            file_name: the name of the file containing the model.
+
+        Returns:
+            a model manager instance for interval-based models.
+        """
+        model_repository = IntervalModelRepository(
+            directory=directory, file_name=file_name, num_models=24 * 60 // 15
+        )
+        return cls(model_repository=model_repository)
+
+    def __getitem__(self, key: Any) -> T:
+        """Get a specific loaded model using the subscript operator.
+
+        Args:
+            key: the key to identify the model.
+
+        Returns:
+            the model instance corresponding to the key.
+        """
+        return self.get_model(key)
+
+    def get_model(self, key: Any = None) -> T:
+        """Get a specific loaded model.
+
+        Args:
+            key: the key to identify the model.
+
+        Raises:
+            KeyError: if the key is invalid or it is not provided for
+                multi-model retrieval.
+
+        Returns:
+            the model instance corresponding to the key.
+        """
+        if key is None and len(self._models) != 1:
+            raise KeyError("No key provided for multi-model retrieval")
+
+        if key is None:
```

Comment:
This could be combined with the previous `if`.

Reply:
will adjust it